### ISSUE TITLE
I cannot change the start time

### ISSUE THREAD

**Issue by Lazza** [wdullaer/MaterialDateTimePicker#242]
I am using two `TimePickerDialog` in my application. Since they have to show up multiple times, I carefully store them in the fragment manager and then retrieve them so I do not re-create them from scratch each time.

I am using the following code to change the allowed interval:

    protected void adjustTimePicker(Date min, Date max, TimePickerDialog dialog) {
        dialog.setMinTime(0, 0, 0);
        dialog.setMaxTime(23, 59, 59);

        Calendar calendar = Calendar.getInstance();
        calendar.setTime(min);
        dialog.setMinTime(calendar.get(Calendar.HOUR_OF_DAY), calendar.get(Calendar.MINUTE), 0);
        calendar.setTime(max);
        dialog.setMaxTime(calendar.get(Calendar.HOUR_OF_DAY), calendar.get(Calendar.MINUTE), 0);
        calendar.setTime(min);
        dialog.setStartTime(calendar.get(Calendar.HOUR_OF_DAY), calendar.get(Calendar.MINUTE), 0);
    }

This changes the min time and the max time perfectly. However, `setStartTime` does not work. When I display the `TimePickerDialog` again (right after calling the method above), the interval is correct but the start time is not right. For instance, if I set `14:00` as min and `16:00` as max, the start time might be `16:00`.

Is `setStartTime` supposed to behave this way?

**Comment by wdullaer** [comment]
The library assumes that you're not reusing the dialogs. The overhead of creating a new one is small enough to not be relevant.

If you do reuse them, some functions might work, others might not.

**Comment by Lazza** [comment]
I see. Then it is not clear to me what `setStartTime` is supposed to do. Is it for using it before _displaying_ the dialog?

Anyway thank you for your reply, I have changed my app to re-create dialogs each time.

**Comment by wdullaer** [comment]
Fair point.
I can't remember why I added it (or even if I added it). I'd have to comb through the git history.

If the Dialogs were to be reusable, I wouldn't have named it like that. Some people would have expected their old selections to be the default one, others would want to see their init time.

If you're really intend on reusing them, you just need to add a method that allows to set the current selected time.
The other parts will handle reuse fine, as a side effect of being tolerant to orientation changes.
